packages/grid/_modules_/grid/components/toolbar/GridToolbarExport.tsx

```diff
@@ -7,31 +7,44 @@ import MenuItem from '@material-ui/core/MenuItem';
 import { isHideMenuKey, isTabKey } from '../../utils/keyboardUtils';
 import { GridApiContext } from '../GridApiContext';
 import { GridMenu } from '../menu/GridMenu';
-import { GridExportOption, GridExportFormat, GridExportConfiguration } from '../../models';
+import { GridExportCsvOptions } from '../../models/gridExport';
 
-type GridToolbarExportProps = ButtonProps & { exportConfiguration?: GridExportConfiguration };
+interface GridExportFormatCsv {
+  format: 'csv';
+  formatOptions?: GridExportCsvOptions;
+}
+
+type GridExportFormatOption = GridExportFormatCsv;
+
+type GridExportOption = GridExportFormatOption & {
+  label: React.ReactNode;
+};
+
+export interface GridToolbarExportProps extends ButtonProps {
```

Comment:
When `GridToolbarExportProps` is the extended interface of `ButtonProps`, I get a missing properties error. Using a type instead of an interface solves this problem.

Reply:
Which error do you get?

Reply:
It sounds like https://github.com/mui-org/material-ui/issues/15827

Reply:
> Type '{ csvOptions: { fileName: string; utf8WithBom: true; }; }' is missing the following properties from type 'Pick<GridToolbarExportProps, "id" | "form" | "slot" | "style" | "title" | "children" | "classes" | "onError" | "aria-label" | "aria-labelledby" | "className" | ... 274 more ... | "TouchRippleProps">': innerRef, buttonRefts(2739)

Reply:
OK, great. Could we add a TypeScript test (`spec.tsx`) to enforce this constraint. This might change in MUI core and break the solution we put in place here.

Reply:
For the solution, we can likely depend on https://next.material-ui.com/guides/typescript/#usage-of-component-prop

Reply:
@oliviertassinari I didn't get any error using the `csvOptions`. Here's a CodeSandbox using it: https://codesandbox.io/s/material-demo-forked-6f54y?file=/demo.js 

How mui-org/material-ui#15827 is related to this?

Reply:
After updating the rest of the packages in my local project to the latest version, the problem is gone.